docs/design/kubefile_alpha.md

```diff
@@ -0,0 +1,111 @@
+or private cloud, or on bare metal. Besides, Kubefile brings much more efficiency
+in distributed application delivery lifecycle.
+
+Without Kubefile, IT engineers take quite complicated steps to run distributed
+application(s). Taking a three-node etcd cluster without Helm as an example, IT engineers must
+prepare three nodes which meet the reprequisites of etcd installation, download
+and configure essential dependencies of etcd, and start to install and construct
+an etcd cluster. If etcd cluster is to be installed by Helm on an existing Kubernets,
+it could be shorten steps above without concerning about reprequisites and manually
+startup trigger. If there is no existing Kubernetes, Kubefile and ClusterImage are
+the best to choice to setup etcd cluster with one command just in minutes.
+
+## Concept Origin
+
+Kubefile has quite similar concept with Dockerfile. And Kubefile is indeed inpired
+by Dockerfile. Actullay, Dockerfile encapsulates single-host application(s) into a single-host
+box(Docker image). While Kubefile encapsulates cluster-level distributed
+application(s) into a cluster-level box(ClusterImage).
+
+Docker uses Dockerfile to build Docker image, and run Docker container with Docker image.
+
+sealer uses Kubefile to build ClusterImage, and run Kubernetes cluster with ClusterImage.
+
+Docker tackles almost all of delivery issues of single-host application(s). sealer expands
+the concept to cluster level, and mostly focus on the perspetives of distributed application(s).
+To be specific, since Kubefile contains a Kubernetes cluster inside of itself by deault and
+Kubernetes has powerful ability to manage containers from Dockerfiles, we can say that
+Kubefile's concept in built on the basis of Dockerfile, Kubernetes.
+
+## Kubefile Command Syntax
+
+The recommend commands in Kubefile are as following:
+```
+FROM [scratch/kubernetes:v1.19.8] (specify a base image)
+CNI  (specify a cni to be deployed in cluster)
+CSI  (specify a csi to be deployed in cluster)
+APP  (declare apps to install)
+LAUNCH [applist]
+```
+`CNI`, `LAUNCH` are allowed to be declared once.
+
+`APP`, `CSI` are allowed to be declared many times.
+
+The further introductions for command as follows:
+
+### FROM
+`FROM` allows user to define the base image. User is able to do some incremental operations on
+base image.
+
+base image could be any of cluster image or scratch.
+
+### CNI
+
+`CNI` allows user to define which cni to be installed over Kubernetes.
+
+Some behaviors are available:
+
+`CNI --type calico / flannel --cidr ...` a easiest way to deploy cni, sealer will support several types of cni.
+
+`CNI --source [path / https://...]`. `path` is a relative path refers to local build context, this can be a directory
+contains multiple yaml or a single yaml for deploying cni. `https://...` is a remote address for downloading related files to deploy cni.
+
+`[]` means user could declare several sources
```

Comment:
I think we could only declare CNI once. So we could not support [] which means multiple sources.